File: SDKs/googleapis.com/dfareporting/v3.3/go/pkg/models/shared/mobileapp.go
package shared

type MobileAppDirectoryEnum string

const (
	MobileAppDirectoryEnumUnknown         MobileAppDirectoryEnum = "UNKNOWN"
	MobileAppDirectoryEnumAppleAppStore   MobileAppDirectoryEnum = "APPLE_APP_STORE"
	MobileAppDirectoryEnumGooglePlayStore MobileAppDirectoryEnum = "GOOGLE_PLAY_STORE"
)

// MobileApp
// Contains information about a mobile app. Used as a landing page deep link.
type MobileApp struct {
	Directory     *MobileAppDirectoryEnum `json:"directory,omitempty"`
	ID            *string                 `json:"id,omitempty"`
	Kind          *string                 `json:"kind,omitempty"`
	PublisherName *string                 `json:"publisherName,omitempty"`
	Title         *string                 `json:"title,omitempty"`
}

File: SDKs/googleapis.com/dfareporting/v3.4/typescript/src/sdk/models/shared/mobileapp.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";

export enum MobileAppDirectoryEnum {
    Unknown = "UNKNOWN"
,    AppleAppStore = "APPLE_APP_STORE"
,    GooglePlayStore = "GOOGLE_PLAY_STORE"
}


// MobileApp
/** 
 * Contains information about a mobile app. Used as a landing page deep link.
**/
export class MobileApp extends SpeakeasyBase {
  @Metadata({ data: "json, name=directory" })
  directory?: MobileAppDirectoryEnum;

  @Metadata({ data: "json, name=id" })
  id?: string;

  @Metadata({ data: "json, name=kind" })
  kind?: string;

  @Metadata({ data: "json, name=publisherName" })
  publisherName?: string;

  @Metadata({ data: "json, name=title" })
  title?: string;
}
